__tests__/user/user.query.test.js

```diff
@@ -0,0 +1,181 @@
+                            tokenPass
+                        } 
+                    }
+                }
+            `
+        })
+
+        userId = register.data.registerUser._id
+
+        const authRes  = await client.mutate({
+            mutation: gql`
+                mutation {       
+                    loginUser(user: {
+                        email: "test.email@gmail.com"
+                        password: "12345678Te"
+                    }) {
+                        token
+                    }
+                }
+            `
+        })
+        token = authRes.data.loginUser.token
+    });
+
+    afterAll(async () => {
+        await client.mutate({
+            mutation: gql`
+                mutation($userId: ID!) {       
+                    deleteUser(id: $userId) {
+                        _id
+                    }
+                }
+            `,
+            variables: {
+                userId
+            }
+        })
+    });
+
+    test('should recive all users', async () => {
+        const res = await client.query({
+            query: gql`
+                query {       
+                    getAllUsers{
+                        email
+                        firstName
+                        lastName
+                        phoneNumber
+                    }
+                }
+            `
+        })
+         expect(res.data.getAllUsers).toContainEqual({
+            "firstName": "Test" ,
+            "lastName": "User",
+            "email": "test.email@gmail.com",
+            "phoneNumber": null        
+        });
+
+    })
+
+    test('should recive user by token', async () => {
+        const res = await client.query({
+            query: gql`
+                query {       
+                getUserByToken {
+                    email
+                    firstName
+                    lastName
+                    phoneNumber
+                    purchasedProducts
+                    role
+                    orders
+                    wishlist
+                    credentials{
+                        source
+                    }
+                    address{
+                        country
+                        city
+                        street
+                        appartment
+                        buildingNumber
+                    }
+                }
+            }
+            `,
+            context: {
+                headers: {
+                    token
+                }
+            }
+        })
+
+        expect(res.data.getUserByToken).toHaveProperty(
+            'firstName', 'Test'
+        );
+        expect(res.data.getUserByToken).toHaveProperty(
+            'lastName', 'User'
+        );
+        expect(res.data.getUserByToken).toHaveProperty(
+            'email', 'test.email@gmail.com'
+        );
+        expect(res.data.getUserByToken).toMatchSnapshot()
+    })
+
+    test('should recive user by id', async () => {
```

Comment:
Please, add a test that checks that proper errors are returned if the client provides the invalid `userId`

Reply:
fixed